Classes/EventStore/Storage/Doctrine/DoctrineEventStorage.php

```diff
@@ -253,23 +273,63 @@ public function setup()
             return $result;
         }
         if ($tableExists) {
-            $result->addNotice(new Notice('Table "%s" (already exists)', null, [$this->eventTableName], 'Skipping'));
-            return $result;
+            $result->addNotice(new Notice('Table "%s" (already exists)', null, [$this->eventTableName]));
+        } else {
+            $result->addNotice(new Notice('Creating database table "%s" in database "%s" on host %s....', null, [$this->eventTableName, $this->connection->getDatabase(), $this->connection->getHost()]));
         }
-        $result->addNotice(new Notice('Creating database table "%s" in database "%s" on host %s....', null, [$this->eventTableName, $this->connection->getDatabase(), $this->connection->getHost()]));
-
-        $schema = $this->connection->getSchemaManager()->createSchema();
-        $toSchema = clone $schema;
 
-        EventStoreSchema::createStream($toSchema, $this->eventTableName);
+        $fromSchema = $this->connection->getSchemaManager()->createSchema();
+        $schemaDiff = (new Comparator())->compare($fromSchema, $this->createEventStoreSchema());
 
+        $statements = $schemaDiff->toSaveSql($this->connection->getDatabasePlatform());
+        if ($statements === []) {
+            $result->addNotice(new Notice('Table schema is up to date, no migration required'));
+            return $result;
+        }
         $this->connection->beginTransaction();
-        $statements = $schema->getMigrateToSql($toSchema, $this->connection->getDatabasePlatform());
         foreach ($statements as $statement) {
             $result->addNotice(new Notice('<info>++</info> %s', null, [$statement]));
             $this->connection->exec($statement);
         }
         $this->connection->commit();
         return $result;
     }
+
+    /**
+     * Creates the Doctrine schema to be compared with the current db schema for migration
+     *
+     * @return Schema
+     */
+    private function createEventStoreSchema()
```

Comment:
Any reason to have the schema moved from the dedicated file? I found it better to contain the schema specification in an own file

Reply:
There used to be more functionality in the `Schema` folder and in this file. Now there's only one static method left that is invoked in a single place.
What's the advantage of a separate file for this one method?

Reply:
Less "visual debt" :P Nah, really, I just prefer to have different concerns separated in different locations. Schema is a different concern than how the EventStorage is implemented IMO.

Reply:
Personally, I like the new way. Maybe it's just a matter of taste.